runner/actions-runner-dind-rootless.dockerfile

```diff
@@ -0,0 +1,139 @@
+FROM ubuntu:20.04
+
+# Target architecture
+ARG TARGETPLATFORM=linux/amd64
+
+# GitHub runner arguments
+ARG RUNNER_VERSION=2.294.0
+
+# Docker and Docker Compose arguments
+ENV CHANNEL=stable
+ARG COMPOSE_VERSION=v2.6.0
+
+# Dumb-init version
+ARG DUMB_INIT_VERSION=1.2.5
+
+# Other arguments
+ARG DEBUG=false
+
+# Set environment variables needed at build
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt update -y \
+    && apt-get install -y software-properties-common \
+    && add-apt-repository -y ppa:git-core/ppa \
+    && apt-get update -y \
+    && apt-get install -y --no-install-recommends \
+    build-essential \
+    curl \
+    ca-certificates \
+    dnsutils \
+    ftp \
+    fuse-overlayfs \
```

Comment:
I had this PR tested on our self hosted runners infra in GKE.  Our GKE hosts run Ubuntu and we found fuse-overlayfs not working very well on Ubuntu (super slow, deadlocking, etc.).  Everything is working as expected when using default overlay filesystem.

Reply:
@isarkis, I think it's a bit out of scope for this PR - I superimposed my work on this topic from my demo repository onto this one for the PR.  I don't run `fuse-overlayfs` on my own images either, for much the same reason.

Reply:
Then, let's remove fuse-overlayfs from the runner/actions-runner-dind-rootless.dockerfile.  Per https://docs.docker.com/engine/security/rootless/, overlay2 is the recommended file system on Ubuntu.